app/models/print_job.rb

```diff
@@ -4,45 +4,121 @@ class PrintJob # rubocop:todo Style/Documentation
   include ActiveModel::Model
 
   attr_reader :number_of_copies
-  attr_accessor :labels, :printer_name, :label_template
+  attr_accessor :printer, :printer_name, :label_templates_by_service, :labels, :labels_sprint
 
-  validates :printer_name, :label_template, :number_of_copies, :labels, presence: true
+  # Add printer/ labels_sprint below?
+  validates :printer_name, :label_templates_by_service, :number_of_copies, :labels, presence: true
 
-  def execute # rubocop:todo Metrics/MethodLength
+  def execute
     return false unless valid?
 
-    begin
-      job = PMB::PrintJob.new(
-        printer_name: printer_name,
-        label_template_id: label_template_id,
-        labels: { body: all_labels }
-      )
-      if job.save
-        true
-      else
-        errors.add(:print_server, job.errors.full_messages.join(' - '))
-        false
-      end
-    rescue JsonApiClient::Errors::ConnectionError
-      errors.add(:pmb, 'PrintMyBarcode service is down')
+    case printer.print_service
+    when 'PMB'
+      print_to_pmb
+    when 'SPrint'
+      print_to_sprint
+    else
+      errors.add(:base, "Print service #{printer.print_service} not recognised.")
       false
     end
   end
 
+  def print_to_pmb # rubocop:todo Metrics/MethodLength
+    job = PMB::PrintJob.new(
+      printer_name: printer_name,
+      label_template_id: pmb_label_template_id,
+      labels: { body: (labels * number_of_copies) }
+    )
+    if job.save
+      true
+    else
+      errors.add(:print_server, job.errors.full_messages.join(' - '))
+      false
+    end
+  rescue JsonApiClient::Errors::ConnectionError
+    errors.add(:pmb, 'PrintMyBarcode service is down')
+    false
+  end
+
+  def print_to_sprint
+    # labels structure is like this:
+    # "labels"=>[
+    #   {
+    #     "main_label"=>{
+    #       "top_left"=>"29-OCT-2020",
+    #       "bottom_left"=>"DN9000214K",
+    #       "top_right"=>"DN9000211H",
+    #       "bottom_right"=>"Duplex-Seq LDS AL Lib",
+    #       "barcode"=>"DN9000214K"
+    #     },
+    #     "extra_label"=>{
+    #       "top_left"=>"29-OCT-2020",
+    #       "bottom_left"=>"DN9000214K",
+    #       "top_right"=>"DN9000211H",
+    #       "bottom_right"=>"Duplex-Seq LDS AL Lib extra",
+    #       "barcode"=>"DN9000214K"
+    #     }
+    #   },
+    #   {
+    #     "main_label"=>{
+    #       "top_left"=>"29-OCT-2020",
+    #       "bottom_left"=>"DN9000214K",
+    #       "top_right"=>"DN9000211H",
+    #       "bottom_right"=>"Duplex-Seq LDS Lig",
+    #       "barcode"=>"DN9000214K-LIG"
+    #     }
+    #   },
+    # ]
```

Comment:
Would be good to move the above 'labels structure' comment to within the `print_to_pmb` method, as that's where it is used. 

The 'labels_sprint structure' one below is good where it is.